core/src/core/block.rs

```diff
@@ -488,9 +452,10 @@ impl Block {
 
 		// collect all the inputs, outputs and kernels from the txs
 		for tx in txs {
-			all_inputs.extend(tx.inputs);
-			all_outputs.extend(tx.outputs);
-			all_kernels.extend(tx.kernels);
+			let tb: TransactionBody = tx.into();
+			all_inputs.extend(tb.inputs);
```

Comment:
Most access appears to be along the pattern of `block.inputs()` or `tx.inputs()` now.
It may feel more consistent if we could do this here also (via an accessor and not a direct field access)?

Reply:
Accessors borrow references to fields, here we need actual values, not references. Direct field access in Rust allows partial move of values from the struct, which is quite unique for Rust. I tried to make this case explicit by type casting, instead of returning `block` filed from the method (which implicitly destroys the tx here, quite unexpected side effect). Another approach would be clone, but I don't like it here because of loop, overhead may be quite big.

Reply:
That makes sense I think. I still don't have an intuitive grasp of this in Rust... I think I get what you are saying.